clients/src/main/java/org/apache/kafka/common/network/ChannelBuilders.java

```diff
@@ -29,16 +32,18 @@ private ChannelBuilders() { }
     /**
      * @param securityProtocol the securityProtocol
      * @param loginType the loginType, it must be non-null if `securityProtocol` is SASL_*; it is ignored otherwise
-     * @param configs client configs
+     * @param config client config
+     * @param listenerName the listenerName if loginType is SERVER or null otherwise
      * @param clientSaslMechanism SASL mechanism if mode is CLIENT, ignored otherwise
      * @param saslHandshakeRequestEnable flag to enable Sasl handshake requests; disabled only for SASL
      *             inter-broker connections with inter-broker protocol version < 0.10
      * @return the configured `ChannelBuilder`
      * @throws IllegalArgumentException if `mode` invariants described above is not maintained
      */
     public static ChannelBuilder clientChannelBuilder(SecurityProtocol securityProtocol,
-            LoginType loginType,
-            Map<String, ?> configs,
+            JaasContext.Type loginType,
```

Comment:
Perhaps `loginType` should be renamed to `contextType` (or `JaasContext.Type` be renamed to `JaasContext.LoginType`? Otherwise it will be confusing if you are not familiar with the old `LoginType`. There are references to `loginType` in several places.